File: 0_3.leetcode/java/Roman_To_Integer.java
public class Roman_To_Integer {
    public int romanToInt(String s) {
        int[] numbers = new int[s.length()];
        for (int i=0;i<s.length();i++) {
            switch (s.charAt(i)) {
                case 'I' -> numbers[i] = 1;
                case 'V' -> numbers[i] = 5;
                case 'X' -> numbers[i] = 10;
                case 'L' -> numbers[i] = 50;
                case 'C' -> numbers[i] = 100;
                case 'D' -> numbers[i] = 500;
                case 'M' -> numbers[i] = 1000;
            }
        }

        int answer = numbers[numbers.length - 1];
        for (int i=0;i<numbers.length - 1;i++) {
            if (numbers[i] < numbers[i + 1]) {
                answer -= numbers[i];
            } else {
                answer += numbers[i];
            }
        }

        return answer;
    }
}

File: 0_3.leetcode/kotlin/src/main/kotlin/Roman_To_Integer.kt
fun main() {
    val result = romanToInt("III")
    println("result = $result")
}

fun romanToInt(s: String): Int {
    val numbers = IntArray(s.length)
    s.forEachIndexed { index, c ->
        val number = when(c) {
            'I' -> 1
            'V' -> 5
            'X' -> 10
            'L' -> 50
            'C' -> 100
            'D' -> 500
            'M' -> 1000
            else -> throw Exception("Unexpected letter $c")
        }
        numbers[index] = number
    }
    var answer = numbers[numbers.size - 1]
    for (i in 0 until numbers.size - 1) {
        if (numbers[i] < numbers[i + 1]) {
            answer -= numbers[i]
        } else {
            answer += numbers[i]
        }
    }
    return answer
}